lib/Controller/UsersController.php

```diff
@@ -395,7 +442,26 @@ public function create($username, $password, array $groups= [], $email='') {
 		}
 
 		try {
+			$setPasswordBeforeLogin = true;
+			if (($password === '') && ($email !== '')) {
+				/**
+				 * Generate a random password as we are going to have this
+				 * use one time. The new user has to reset it using the link
+				 * from email.
+				 */
+				$event = new GenericEvent();
+				$this->eventDispatcher->dispatch('OCP\User::createPassword', $event);
+				if ($event->hasArgument('password')) {
+					$password = $event->getArgument('password');
+				} else {
+					$password = $this->secureRandom->generate(20);
+				}
+				$setPasswordBeforeLogin = false;
+			}
 			$user = $this->userManager->createUser($username, $password);
+			if ($setPasswordBeforeLogin === false) {
+				$this->config->setUserValue($username, 'owncloud', 'disable_password_reset', 'true');
```

Comment:
this is not clear, why disable the password reset feature ? a user should still be able to enter an invalid password in the login field (in case they didn't get an email) and ask for a password reset

I'm only interpreting this based on the name "disable_password_reset". If it means something different then it needs a better name.